pkg/transport/keepalive_listener.go

```diff
@@ -23,15 +23,10 @@ import (
 
 // NewKeepAliveListener returns a listener that listens on the given address.
 // http://tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html
-func NewKeepAliveListener(addr string, scheme string, info TLSInfo) (net.Listener, error) {
-	l, err := net.Listen("tcp", addr)
-	if err != nil {
-		return nil, err
-	}
-
+func NewKeepAliveListener(l net.Listener, scheme string, info TLSInfo) (net.Listener, error) {
```

Comment:
So this patch is doing the same thing as the old code but it's passing a `Listener` into `NewKeepAliveListener` instead of expecting it to create one for you? There should be a comment here about why this is important.